pkg/autoscaler/stats_scraper.go

```diff
@@ -127,11 +145,41 @@ func (s *ServiceScraper) Scrape() (*StatMessage, error) {
 		return nil, nil
 	}
 
-	stat, err := s.sClient.Scrape(s.url)
-	if err != nil {
-		return nil, err
+	sampleSize := s.samClient.SampleSize(readyPodsCount)
+	if sampleSize == 0 {
+		return nil, nil
+	}
+
+	var avgCon float64
+	var avgProxiedCon float64
+	var reqC int32
+	var proxiedReqC int32
+	var sussC float64
+	var errRec error
+	for i := 0; i < sampleSize; i++ {
+		stat, err := s.sClient.Scrape(s.url)
```

Comment:
Yeah we need to.

Instead of making `StatsScraper.Scrape` wait for returning one aggregated `StatMessage` then pass it to the stats channel, how about passing the stats channel as a parameter of `StatsScraper.Scrape` and let `StatsScraper.Scrape` send as many stats as it wants to the channel? In this way, we don't need extra aggregation since the autoscaler already handles aggregation based on interval buckets.

Reply:
Changed to what I proposed. PTAL.